core/src/main/java/com/sequenceiq/cloudbreak/service/stack/DefaultStackService.java

```diff
@@ -154,6 +154,12 @@ public Stack create(CbUser user, Stack stack) {
         } else {
             try {
                 savedStack = stackRepository.save(stack);
+                subnetRepository.save(stack.getAllowedSubnets());
+                Set<InstanceGroup> instanceGroups = stack.getInstanceGroups();
+                for (InstanceGroup group : instanceGroups) {
+                    group.setStack(savedStack);
+                }
+                instanceGroupRepository.save(instanceGroups);
```

Comment:
That would be more elegant if an instance group related service would set the stack of instance group and save it by using repository directly.